### ISSUE TITLE
30pin adapter to lightning

### ISSUE THREAD

**Issue by jhjulkun** [Navideck/Romo-X#9]
The app builds and runs, but never finds the dock using the adapter I have.
However, I managed to squeeze in an old iPad (with the 30pin connector), and then it works! 
But obviously the iPad is way to heavy.. 

But in short, which adapters work? 
Mine looks like the Apple 30pin to lightning, but its definitely not working.

**Comment by fotiDim** [comment]
If I understand correctly what you are trying to do is "30pin Romo" -> 30pin to lightning adapter -> lightning iPhone. I have never tried this setup. I have such an adapter but it is non-genuine and the cable was ripped off (need to resolder it). 

Is your adapter genuine Apple? 3rd party adapters tend to be finicky. 

I have experience with a certain 30pin accessory that never worked with a 30pin to lightning adapter so it can be that the same case is with Romo.

**Comment by jhjulkun** [comment]
Correct. Its not a genuine Apple adapter (but looks exactly like one), so thats most likely the problem.
I have not been able to pry the Romo apart (without damaging it in the process), maybe would be easier to solder on a lightning cable directly? (assuming its USB, have not looked at the code either).

Update: managed to pry it apart (partially glued, even though there was four screws in the bottom). 
There are 8 wires going to the 30 pin connector, but since that part is glued I can't trace which pins they go to. 
But means I should be able to connect the iPad (by mechanically disabling the tilt servo) and test the SDK until I find an adapter that works for lightning iPhones.

Update2: think I toasted it, motors are no longer moving... Ah well.

**Comment by fotiDim** [comment]
I managed to fix my adapter. However Romo does not react at all when using it. It does not light up at all. When trying the same Romo with an iPhone 4 (30 pin) it works just fine. So it seems this kind of adapter is not supported. If you find a genuine Apple adapter it is worth a try although I wouldn't have my hopes too high.

Regarding disassembly I have put up a [wiki page](https://github.com/Navideck/Romo/wiki/Disassemble-Romo). Hope it helps. I don't know if it is possible to replace the 30 pin connector with lightning. The connector housing (cylinder shaped piece) is attached to the main board with a connector. It might be possible to get the corresponding part from a broken lightning Romo (3L) and place it on yours if the connector is the same.

You can see it in the picture. Does the connector look like yours?

![IMG_2268](https://user-images.githubusercontent.com/2326415/115990445-914ed300-a5c3-11eb-9392-f86c690e84a0.jpg)

**Comment by jhjulkun** [comment]
Thanks for the help! I think I found the problem: when I tried taking off the blue plastic, I must have pried off one of the SOT23 transistors (bummer I didn't see your wiki before!)
I soldered the transistor back, but its still dead (transistor must be involved with motor power somehow, and it looks like one of the legs of the transistor is yanked off). 
My mainboard is v1.0.14 btw (its slightly larger and looks different than your board), so wonder if it supports lightning anyway? I was hoping to use it as a chassis for a robotics project I had in mind, but since I broke it I think I'll go back to the other chassis I had in mind. Bummer!
(or maybe use the motor&battery with another motor board)

**Comment by fotiDim** [comment]
I also made a small damage when I opened mine for the first time. You can see I have put some hot glue in the picture.
If it is just a transistor I would just replace it. It is highly likely that it will work again.
Can you post some pictures for reference? I am trying to gather the knowledge in the wiki.

Your project sounds interesting. Do you plan to modify Romo or to use it as is?

**Comment by jhjulkun** [comment]
I think the SOT23 says S21 (super tiny text!), but this seems to be a diode? 
But board connects to all 3 pins, so I think its a transistor after all, 
and one of the pins seems to yanked off the transistor housing... 
So I would have to replace it with something else, I'll try finding one. 

I was hoping to use it with ARKit, some guys here have done some cool stuff:
https://github.com/laanlabs/HomeRobot

https://www.datasheetq.com/datasheet-download/933881/1/Diodes/BAS21-7
![IMG_6616](https://user-images.githubusercontent.com/2847796/115997190-91aa9680-a5e2-11eb-9646-1e429d3b2bfa.jpg)

**Comment by jhjulkun** [comment]
Btw, do you know how many of the Romo3L models were made? 
I have found the 30pin models on eBay, but never seen the 3L.

**Comment by fotiDim** [comment]
Is your Romo a 3A or 3B?

> I think the SOT23 says S21 (super tiny text!), but this seems to be a diode?
> But board connects to all 3 pins, so I think its a transistor after all,
> and one of the pins seems to yanked off the transistor housing...
> So I would have to replace it with something else, I'll try finding one.

It looks like S21. There are more S21 components on the board so I assume it fits the pattern. See if you find the part by itself otherwise look for a dead (or working) Romo.

> I was hoping to use it with ARKit, some guys here have done some cool stuff:
> https://github.com/laanlabs/HomeRobot

Yes, I am aware of this project.

> Btw, do you know how many of the Romo3L models were made?
> I have found the 30pin models on eBay, but never seen the 3L.

They were definitely made in large quantities. I had no trouble finding them a couple of years ago even new. Not sure where you live but in Germany I can find at least [one](https://www.ebay-kleinanzeigen.de/s-anzeige/romo-fahrbarer-robotor-fuer-iphone-neuwertig/1737172809-173-18276) listed.

**Comment by jhjulkun** [comment]
What are the differences for the 3A&3B? 
Didn't know there also are local ebay listings, thanks!

**Comment by fotiDim** [comment]
> What are the differences for the 3A&3B?
> Didn't know there also are local ebay listings, thanks!

Good question. Packaging is one difference. 3A white box. 3B black/green box. Not sure about the technical differences but it seems to be a revision. Both are 30 pin models.

You can find the exact model on the bottom of Romo.

**Comment by jhjulkun** [comment]
3A. Seems like "S21" is used for many surface mount devices, but I think its this, I'll try finding a replacement. 
https://www.mouser.se/datasheet/2/196/Infineon-BSS214N-DS-v02_03-en-1731206.pdf
